game-core/src/main/java/games/strategy/triplea/settings/SelectionComponentFactory.java

```diff
@@ -413,6 +423,130 @@ public void resetToDefault() {
+          new EmailProviderSetting("GMail", "smtp.gmail.com", 587, true),
+          new EmailProviderSetting("Hotmail", "smtp.live.com", 587, true)
+      );
+
+      final JTextField serverField = new JTextField(hostSetting.getValue().orElse(""), 20);
+
+      final JSpinner portSpinner = new JSpinner(new SpinnerNumberModel(
+          (int) portSetting.getValue().orElse(0), 0, 65535, 1));
+
+      final JCheckBox tlsCheckBox = new JCheckBox("SSL/TLS", tlsSetting.getValue().orElse(true));
+
+      final JTextField usernameField = new JTextField(usernameSetting.getValue().orElse(""), 20);
+
+      final JPasswordField passwordField = new JPasswordField(passwordSetting.getValue().orElse(""), 20);
+
+      final JPanel panel = JPanelBuilder.builder()
+          .verticalBoxLayout()
+          .addLeftJustified(new JLabel("Email Server"))
+          .addLeftJustified(serverField)
+          .addLeftJustified(new JLabel("Port"))
+          .addLeftJustified(JPanelBuilder.builder()
+              .horizontalBoxLayout()
+              .addLeftJustified(portSpinner)
+              .addLeftJustified(tlsCheckBox)
+          .build())
+          .addVerticalStrut(5)
+          .addLeftJustified(JButtonBuilder.builder()
+              .title("Presets...")
+              .actionListener(() -> {
+                final JComboBox<EmailProviderSetting> comboBox =
+                    JComboBoxBuilder.builder(EmailProviderSetting.class)
+                        .items(knownProviders)
+                        .build();
+                if (JOptionPane.showConfirmDialog(this.panel.getParent(), JPanelBuilder.builder().add(comboBox).build(),
+                    "Select a Preset", JOptionPane.OK_CANCEL_OPTION) == JOptionPane.OK_OPTION) {
+                  final EmailProviderSetting config = (EmailProviderSetting) comboBox.getSelectedItem();
+                  serverField.setText(config.getHost());
+                  portSpinner.setValue(config.getPort());
+                  tlsCheckBox.setSelected(config.isEncrypted());
+                }
+              })
+            .build())
+          .addLeftJustified(new JLabel("Username"))
+          .addLeftJustified(usernameField)
+          .addLeftJustified(new JLabel("Password"))
+          .addLeftJustified(passwordField)
+          .build();
+
+      @Override
+      public JComponent getUiComponent() {
+        return panel;
+      }
+
+      @Override
+      public Map<GameSetting<?>, Object> readValues() {
+        final char[] password = passwordField.getPassword();
+
+        final Map<GameSetting<?>, Object> map = ImmutableMap.<GameSetting<?>, Object>builder()
+            .put(hostSetting, Strings.emptyToNull(serverField.getText()))
+            .put(portSetting, portSpinner.getValue())
+            .put(tlsSetting, tlsCheckBox.isSelected())
+            .put(usernameSetting, Strings.emptyToNull(usernameField.getText()))
+            .put(passwordSetting, password.length == 0 ? null : new String(password))
```

Comment:
Yeah, as you mentioned in the PR description, this is a problem.  If you're going to keep the protect/unprotect stuff at the `ClientSetting` level, you'll probably need to modify how all the `SaveFunction` stuff works so it doesn't display the unprotected value.  That may require adding a new flag to the UI binding that says whether the value should be displayed or not.  However, that, in and of itself, seems to smell a bit.